# We begin with classification, so we fit a baseline classifier that predicts the mode

In [6]:
import pandas as pd
data = "../../data/rolling_feature_engineered_tier_1_games.csv"
df = pd.read_csv(data)
df

,previous_10_game_team1_player1_average_kills,previous_10_game_team1_player1_average_deaths,previous_10_game_team1_player1_average_assists,previous_10_game_team1_player1_average_adr,previous_10_game_team1_player1_average_kast,previous_10_game_team1_player1_average_kddiff,previous_10_game_team1_player2_average_kills,previous_10_game_team1_player2_average_deaths,previous_10_game_team1_player2_average_assists,previous_10_game_team1_player2_average_adr,...,team1_player1_id,team1_player2_id,team1_player3_id,team1_player4_id,team1_player5_id,team2_player1_id,team2_player2_id,team2_player3_id,team2_player4_id,team2_player5_id
0,16.000000,15.000000,4.000000,75.000000,70.000000,1.000000,16.000000,15.000000,4.000000,75.000000,...,553.0,39.0,612.0,163.0,457.0,739.0,1443.0,65.0,2049.0,209.0
1,12.000000,17.000000,4.000000,59.100000,62.500000,-5.000000,13.000000,11.000000,2.000000,52.600000,...,553.0,39.0,612.0,163.0,457.0,739.0,1443.0,65.0,2049.0,209.0
2,13.500000,16.500000,4.500000,66.250000,71.750000,-3.000000,18.500000,11.000000,4.000000,82.550000,...,553.0,39.0,612.0,163.0,457.0,739.0,1443.0,65.0,2049.0,209.0
3,14.612903,14.645161,4.258065,69.832258,71.535484,-0.032258,14.612903,14.645161,4.258065,69.832258,...,3202.0,1420.0,3168.0,607.0,3032.0,3836.0,1886.0,2588.0,1778.0,4567.0
4,13.000000,5.000000,1.000000,81.300000,86.700000,8.000000,11.000000,6.000000,4.000000,72.100000,...,3202.0,1420.0,3168.0,607.0,3032.0,3836.0,1886.0,2588.0,1778.0,4567.0
...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...,...
5534,15.400000,14.800000,4.900000,73.770000,77.820000,0.600000,14.200000,12.900000,3.900000,66.030000,...,2988.0,266.0,7528.0,6900.0,16113.0,1401.0,169.0,1834.0,2934.0,7201.0
5535,16.600000,15.500000,5.000000,76.570000,76.160000,1.100000,14.900000,13.400000,3.900000,66.480000,...,2988.0,266.0,7528.0,6900.0,16113.0,1401.0,169.0,1834.0,2934.0,7201.0
5536,15.400000,15.900000,4.600000,73.230000,73.160000,-0.500000,14.800000,13.600000,3.700000,66.540000,...,2988.0,266.0,7528.0,6900.0,16113.0,1401.0,169.0,1834.0,2934.0,7201.0
5537,13.300000,13.400000,4.500000,67.660000,73.790000,-0.100000,10.700000,13.700000,5.800000,57.120000,...,1401.0,169.0,1834.0,2934.0,7201.0,19.0,550.0,1549.0,1443.0,146.0


# Baseline accuracy rate

We always do a 80/20 chronological split

In [8]:
import math
y = df.team1_win
split_point = math.ceil(len(y) * 0.8)
y_train = y[:split_point]
y_test = y[split_point:]

In [9]:
((y_test == y_train.mode()[0]).sum())/len(y_test)

np.float64(0.5582655826558266)

In [10]:
((y_train == y_train.mode()[0]).sum())/len(y_train)

np.float64(0.5464801444043321)

Test and train are roughly equally distributed. Baseline is 55.8% accurate on test, 54.6% accurate on train.